internal/fdo.go

```diff
@@ -419,16 +461,16 @@ func findOneAppFromNames(f desktop.ApplicationProvider, names ...string) desktop
 	return nil
 }
 
-func appendAppIfExists(apps []desktop.AppData, app desktop.AppData) []desktop.AppData {
```

Comment:
As mentioned above I'm pretty sure that this chunk should revert to the more complete API listing app data.